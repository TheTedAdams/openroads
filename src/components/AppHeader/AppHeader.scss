@import '../../styles/colors';

.AppHeader {
  grid-column: 1 / span 2;
  grid-row: 1;
  background-color: $gray;
  padding: 1rem 1rem 0 1rem;
  color: white;
  display: flex;

  &-logo {
    height: 5rem;
    width: auto;
    display: block;
  }

  &-title {
    font-size: 1.5em;
    flex: 1;
  }

  &-title-unread-count {
    background: $red;
    color: white;
    border-radius: 2px;
    border: solid 1px white;
    font-size: 1rem;
    line-height: 1rem;
    height: 1rem;
    padding: 0.2rem 0.3rem;
    margin-left: 0.3rem;
  }
}
