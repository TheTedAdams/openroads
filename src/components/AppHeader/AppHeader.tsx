import './AppHeader.scss';

import React, { FC } from 'react';
import logo from './logo.svg';

interface AppHeaderProps {
  isLoading: boolean;
  unreadCount: number;
}

export const AppHeader: FC<AppHeaderProps> = ({ isLoading, unreadCount }) => {
  return (
    <header className="AppHeader">
      {/* Just wanted to call out that having padding embedded in the svg is probably not ideal.
       ** I would get the padding removed from the svg and then put align-items: center on the heading instead of the
       ** current mixture of padding and margin (and padding embedded inside the svg). This would make it much more easy
       ** adjust things later.
       */}
      <img src={logo} className="AppHeader-logo" alt="logo" />
      <h1 className="AppHeader-title">Notes Viewer Test App</h1>
      <div>
        Unread:
        <span className="AppHeader-title-unread-count">
          {/* Adding .toLocaleString() because this gives us great freebie handling for 4+ digit numbers */}
          {isLoading ? '-' : (unreadCount ?? 0).toLocaleString()}
        </span>
      </div>
    </header>
  );
};
