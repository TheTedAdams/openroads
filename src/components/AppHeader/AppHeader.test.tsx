import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { AppHeader } from './AppHeader';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AppHeader isLoading={false} unreadCount={0} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Renders like the snapshot', () => {
  const tree = renderer
    .create(<AppHeader isLoading={false} unreadCount={0} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
