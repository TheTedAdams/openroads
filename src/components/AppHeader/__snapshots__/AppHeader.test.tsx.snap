// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders like the snapshot 1`] = `
<header
  className="AppHeader"
>
  <img
    alt="logo"
    className="AppHeader-logo"
    src="logo.svg"
  />
  <h1
    className="AppHeader-title"
  >
    Notes Viewer Test App
  </h1>
  <div>
    Unread:
    <span
      className="AppHeader-title-unread-count"
    >
      0
    </span>
  </div>
</header>
`;
